lua/browse/utils.lua

```diff
@@ -79,18 +81,29 @@ end
 
 -- a generic searching closure util
 M.callback_search = function(custom_fn, opts)
-    return function()
+    return function(visual_text)
+        opts.visual_text = visual_text
         M.search(custom_fn, opts)
     end
 end
 
 -- a generic searching for a format
 M.format_search = function(format, opts)
-    return function()
+    return function(visual_text)
+        opts.visual_text = visual_text
         M.search(function(input)
             return string.format(format, input)
         end, opts)
     end
 end
 
+M.get_visual_text = function()
```

Comment:
write some documentation in the form of comments for the function about what is being done in the process.